openedx/core/djangoapps/dark_lang/middleware.py

```diff
@@ -126,13 +137,9 @@ def _activate_preview_language(self, request):
         auth_user = request.user.is_authenticated()
 
         if auth_user:
-            preview_lang = get_user_preference(request.user, LANGUAGE_KEY) or settings.LANGUAGE_CODE
+            preview_lang = get_user_preference(request.user, LANGUAGE_KEY) or check_user_browser_lang(request)
```

Comment:
What were your DarkLang released languages look like during the demo? Looks like your fix will work only if there are 'ru', 'be' and 'uk' in that config. Because you modified the `_activate_preview_language` method, but first, the `_clean_accept_headers` method applied. That method cleans up the language headers and then you try to compare that header language to the `LANGUAGES_MAPPING`. I think this might be a problem.